### PR TITLE
#122: support multi-turn datasets(chase\sparc\cosql), and merge all data together

### PR DESCRIPTION
**Pull request by oushu1zhangxiangxuan1**
Something not sure:
     Is "instruction" necessary for non-first interaction in multi-turn data?

https://github.com/eosphoros-ai/DB-GPT-Hub/pull/123/files#r1382805358

### CODE REVIEW

File: dbgpt_hub/data_process/sql_data_process.py [hunk 1]
```diff
@@ -84,49 +84,73 @@ def decode_json_file(self, data_file_list, table_file, out_file):
...
-
-        with open(out_file, "w", encoding="utf-8") as s:
-            json.dump(res, s, indent=4, ensure_ascii=False)
+            if data[db_id_name] in db_dict.keys():
+                if is_multiple_turn: #多轮
+                    history = []
+                    for interaction in data["interaction"]:
+                        input = {
+                            "db_id": data[db_id_name],
+                            "instruction": INSTRUCTION_PROMPT.format(db_dict[data[db_id_name]]),
```
Comment (oushu1zhangxiangxuan1) [inline dbgpt_hub/data_process/sql_data_process.py:95]: Is "instruction" necessary for non-first interaction in multi-turn data?

Comment (wangzaistone) [reply dbgpt_hub/data_process/sql_data_process.py:95]: It is necessary in my opinion , it's better to do a compared expriment

**Review by wangzaistone — COMMENTED** [review]
It's great work to merge more data in this way , has you run pass all code in the default command ?

**Comment by oushu1zhangxiangxuan1** [comment]
> It's great work to merge more data in this way , has you run pass all code in the default command ?

Thanks, it works in my environment.

**Comment by wangzaistone** [comment]
> > It's great work to merge more data in this way , has you run pass all code in the default command ?
> 
> Thanks, it works in my environment.

Could you tell us about your environment ?  we want to make a test ,thanks .
We look forward to inviting you to the WeChat group of contributors in our community. If it is convenient, you can tell me your WeChat ID or abbreviation in a specific WeChat group in our community. Thank you very much for your contribution.

**Review by wangzaistone — APPROVED** [review]

File: dbgpt_hub/data_process/sql_data_process.py (hunk 1 above)
Comment (oushu1zhangxiangxuan1) [reply dbgpt_hub/data_process/sql_data_process.py:95]: > It is necessary in my opinion , it's better to do a compared expriment

Adding an 'instruction' for each turn dose not perform well so far, I'm attempting to conduct an ablation study.

**Comment by oushu1zhangxiangxuan1** [comment]
> > > It's great work to merge more data in this way , has you run pass all code in the default command ?
> > 
> > 
> > Thanks, it works in my environment.
> 
> Could you tell us about your environment ? we want to make a test ,thanks . We look forward to inviting you to the WeChat group of contributors in our community. If it is convenient, you can tell me your WeChat ID or abbreviation in a specific WeChat group in our community. Thank you very much for your contribution.

My data folder structure referring to the config in this PR(my chase folder may diff with the official, but can set my own config in dbgpt_hub/configs/config.py)：
image

**Comment by oushu1zhangxiangxuan1** [comment]
> > > It's great work to merge more data in this way , has you run pass all code in the default command ?
> > 
> > 
> > Thanks, it works in my environment.
> 
> Could you tell us about your environment ? we want to make a test ,thanks . We look forward to inviting you to the WeChat group of contributors in our community. If it is convenient, you can tell me your WeChat ID or abbreviation in a specific WeChat group in our community. Thank you very much for your contribution.

Thanks for inviting! I'm in the "DB-GPT" WeChat group, you can search in the group by my id prefix "zxx448"

**Comment by oushu1zhangxiangxuan1** [comment]
> > > It's great work to merge more data in this way , has you run pass all code in the default command ?
> > 
> > 
> > Thanks, it works in my environment.
> 
> Could you tell us about your environment ? we want to make a test ,thanks . We look forward to inviting you to the WeChat group of contributors in our community. If it is convenient, you can tell me your WeChat ID or abbreviation in a specific WeChat group in our community. Thank you very much for your contribution.

Thanks for inviting! I'm in the "DB-GPT" WeChat group, you can search in the group by my id prefix "zxx448"

**Comment by wangzaistone** [comment]
> > > > It's great work to merge more data in this way , has you run pass all code in the default command ?
> > > 
> > > 
> > > Thanks, it works in my environment.
> > 
> > 
> > Could you tell us about your environment ? we want to make a test ,thanks . We look forward to inviting you to the WeChat group of contributors in our community. If it is convenient, you can tell me your WeChat ID or abbreviation in a specific WeChat group in our community. Thank you very much for your contribution.
> 
> Thanks for inviting! I'm in the "DB-GPT" WeChat group, you can search in the group by my id prefix "zxx448"

Could you tell me about the specific WeChat group ?  I have searched around 9 group ,but  I still did not find  you  by  search prefix "zxx448". May be you are  in the group which I'm not in , my wechat can searched by "dabing3520"  and you search this  id may be more convenient for us ,  there are 13 wechat group totally.

**Comment by Oops322** [comment]
> > > > 以这种方式合并更多数据是很棒的工作，您是否在默认命令中运行传递所有代码？
> > > 
> > > 
> > > 谢谢，它适用于我的环境。
> > 
> > 
> > 您能告诉我们您的环境吗？我们想做一个测试，谢谢。我们期待邀请您加入我们社区的微信贡献者群。如果方便的话，您可以在我们社区的特定微信群中告诉我您的微信ID或缩写。非常感谢您的贡献。
> 
> 我的数据文件夹结构参考了这个 PR 中的配置（我的 chase 文件夹可能与官方不同，但可以在 dbgpt_hub/configs/config.py 中设置我自己的配置）：
> 
> 图像

Hello, could you please tell me how to get a series of gold_chase files in your chase data set? The data set I downloaded from the official chase does not have gold.sql like spider. It is very difficult to understand when I want to evaluate the predict.sql of chase

**Pull request merged**